xarray/tests/test_dask.py

```diff
@@ -172,13 +201,15 @@ def test_pickle(self):
     def test_reduce(self):
         u = self.eager_var
         v = self.lazy_var
-        self.assertLazyAndAllClose(u.mean(), v.mean())
-        self.assertLazyAndAllClose(u.std(), v.std())
-        self.assertLazyAndAllClose(u.argmax(dim="x"), v.argmax(dim="x"))
-        self.assertLazyAndAllClose((u > 1).any(), (v > 1).any())
-        self.assertLazyAndAllClose((u < 1).all("x"), (v < 1).all("x"))
-        with raises_regex(NotImplementedError, "dask"):
-            v.median()
+        with _set_dask_scheduler(CountingScheduler(0)):
+            # None of the methods should trigger compute at this stage.
+            self.assertLazyAndAllClose(u.mean(), v.mean())
```

Comment:
Initially, I was a little confused when I read this because I did not realize that the scheduler was overwritten again inside `assertLazyAnd`.

I would rather write these with something like:
```python
with raise_if_dask_computes():
    result = u.mean()
self.asserLazyAndAllClose(result, v.mean()
```
`raise_if_dask_computes` is more explicit than setting the schedule, and leaving the context manager makes it obvious that it’s restrictions no longer apply.

Reply:
I agree, this would be more self-explanatory, but a lot more to type, if applied to all tests...